### PR TITLE
Added Diamond Target

### PR DESCRIPTION
**Pull request by Febianchi**
Added diamond target, made changes in .gitignore and made changes in generator.cc to randomly place the particles relatively to the diamond's plane.

### CODE REVIEW

**Comment by diemort** [comment]
Please add a image from Geant4 in a comment here to show the simulation code is indeed working

**Comment by Febianchi** [comment]
Simulation image:

simulation_image

File: simulation/construction.cc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+	// Box for lower part of the detector:
+
+	G4Box *solidVacuumLower = new G4Box("solidVacuumLower", 1*m, 1*m, 3*m);
+
+	G4LogicalVolume *logicVacuumLower = new G4LogicalVolume(solidVacuumLower, vacuum, "logicalVacuumLower");
+
+	G4VPhysicalVolume *physVacuumLower = new G4PVPlacement (0, G4ThreeVector(0., -2*m, 5*m), logicVacuumLower, "physVacuumLower", logicWorld, false, 0, true); 
+
+	/*
```
Comment (diemort) [inline simulation/construction.cc]: remove comments

File: simulation/construction.cc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	G4Material *diamond = new G4Material("diamond", 3.515*g/cm3, 1);
+	diamond->AddElement(nist->FindOrBuildElement("C"), 1);
+
+	G4Box *solidDiamondTarget = new G4Box("solidDiamondTarget", 0.5*m, 0.5*m, 0.0001*m); //100 micrometers thick
+
+	G4LogicalVolume *logicalDiamondTarget = new G4LogicalVolume(solidDiamondTarget, diamond, "logicalDiamondTarget");
+
+	G4VPhysicalVolume *physDiamond = new G4PVPlacement(0, G4ThreeVector(0., 0., -0.0201*m), logicalDiamondTarget, "physDiamond", logicWorld, false, 0, true);
+
+	/* Assembly Volumes:
```
Comment (diemort) [inline simulation/construction.cc]: remove comments

File: simulation/physics.cc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include "physics.hh"
+
+MyPhysicsList::MyPhysicsList()
+{
+	RegisterPhysics (new G4EmStandardPhysics());
+	RegisterPhysics (new G4OpticalPhysics());
+
```
Comment (diemort) [inline simulation/physics.cc]: remove blank line

File: simulation/construction.cc (hunk 1 above)
Comment (diemort) [reply simulation/construction.cc]: here

File: simulation/construction.cc (hunk 2 above)
Comment (diemort) [reply simulation/construction.cc]: here

File: simulation/physics.cc (hunk 3 above)
Comment (diemort) [reply simulation/physics.cc]: here

**Review by diemort — CHANGES_REQUESTED** [review]

**Comment by Febianchi** [comment]
Commentaries were removed

**Pull request merged**